docs/semgrep-supply-chain/license-compliance.md

```diff
@@ -135,4 +135,22 @@ Exempted packages appear in the Supply Chain > **Settings** tab.
 
 Package exemptions are currently version-specific. Each version used must be exempted individually.
 
+### Custom dependency exceptions
```

Comment:
Just curious, how does "custom dependency exceptions" interact with "exempting packages"?

Also, do we want to establish a preferred term for dependency / package? When I read the two sections (pretending that I am jumping in from Google) my first thought is "why is one "package" and the other "dependency"?

Reply:
I asked a clarifying question about this above; I think one is a subset (superset) of the other -- either way, I'll update with an explanation.

Good catch on package/dependency; I think I'll update all instances of "package" to "dependencies" since that's what's in the UI.

Reply:
@khorne3 Thanks for asking the clarifying question - curious if you tested the behaviour described here yourself? 

I expect this workflow to do the same thing under the hood as clicking on the "allow license exception button" from the Dependencies page - the reason we added this was because there would be dependencies that aren't present on that list in UI beforehand, and hence there is no way to create an exception for it with manually adding an entry in the DB.

With this workflow, if they are currently on `bitwarden/cli@2023.9.0` and want to upgrade to `bitwarden/cli@2023.9.1` without Semgrep's license policy blocking the PR, they will need to create a custom dependency exception for the package they are upgrading **to**, i.e. `bitwarden/cli@2023.9.1`.

I know it's not a perfect workflow, but it was done as a stop gap for the time-being, while we flush out a better, longer term license policy framework. Happy to provide more context if needed :)

Reply:
When I tested the new functionality, it behaved exactly as the license exception button, so if that's what's expected, we're good to go here (tbh, I think that was part of my confusion -- I was expecting differing behavior). I'll update the text to make it clearer how these two methods are the same/different. Thank you!

Reply:
No problem! I'm happy to review the updated docs when they're ready!

Reply:
@gautambhat please let me know what you think of the updates. thanks!